### DISCUSSION TITLE
GITHUB - Service Accounts

### DISCUSSION THREAD

**Discussion by SarahPrice-USDA** [community/community#139865] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Is there a way to set up a service account in GITHUB? I've reviewed the documentation and cannot find any information.

We have established a GITHUB to JIRA integration with an oAuth app but when we sync it in JIRA it is requiring an account. We do not want this tied to a user account but a service type account. 

Any help would be greatly appreciated.

**Comment by lovrozagar** [comment]
Hi @srprice12 ,

Currently, GitHub doesn’t have a dedicated feature for service accounts like some other platforms do. However, you can achieve a similar setup using a couple of approaches:

GitHub App: Consider creating a GitHub App instead of using OAuth for your integration with JIRA. GitHub Apps operate independently of user accounts and can be granted permissions specifically for the actions you need. This way, you won't be tied to any individual user.

Personal Access Token: If you still prefer to use OAuth and need to tie it to a user, you could create a separate GitHub user account specifically for this purpose. You can then generate a Personal Access Token for that account, which can be used for API access. Just make sure to limit the token's scopes to only what you need.

OAuth Application: If you already have an OAuth application set up, you might need to create a dedicated user account for it. This would allow the app to authenticate without being tied to an individual user’s account.

Unfortunately, there isn't a built-in "service account" option in GitHub at this time, but these methods should help you achieve a similar outcome.

If you have more questions or need further assistance, feel free to ask!
